src/assets/src/validation.ts

```diff
@@ -72,21 +78,32 @@ export interface ValidationResult {
 }
 
 export function validateString (value: string, schema: StringSchema, showRemaining: boolean): ValidationResult {
-    let transformedValue;
+    let transformedValue: string;
     let isInvalid = false;
-    let messages = Array();
+    let messages: string[] = Array();
     try {
-        transformedValue = schema.validateSync(value);
+        const result = schema.validateSync(value);
+        if (result === undefined) {
+            throw Error('validateString should have received a string in the value parameter, but it is undefined.');
```

Comment:
I just felt `validateString` is called across multiple component, the error msg is too generic doesn't give enough info where this error occurred. This is optional, I don't know enough understanding with UI flow to clearly say if this will be confusing or not

Reply:
I'm basically just throwing this error here because the type signature on `schema.validateSync` leaves it to possibly be `undefined`. Our `value` parameter is a string, so I don't think we'll ever get `result` as `undefined`. So basically I'm just doing it to satisfy TypeScript.